### DISCUSSION TITLE
Process multiple cue files via Go

### DISCUSSION THREAD

**Discussion by gilcrest** [cue-lang/cue#1623] [Q&A] [answered]
Newbie question here... I have 2 cue files, one for my schema `schema.cue` and constraints and another for my local config data `local.cue`.

If I use the cue cli to validate these files, all is good: `cue vet -c schema.cue local.cue`

Knowing these are valid cue files, I would like to then process these files and use the configuration in my Go program.

Looking at some examples, I see I can take a file, get its bytes, compile those to a value and then parse from there.

```go
b, err := os.ReadFile("path")
if err != nil {
	return nil, err
}

ctx := cuecontext.New()
v := ctx.CompileBytes(b)
```

Given my use case is multiple files, do I need to merge the two files using io.Copy and get their combined bytes or does CUE have any helpers that I'm not finding for something like this?

Also, just to be clear - I am hoping to just parse my program configuration straight from cue files, unless you think that's a bad idea and I should use CUE to export to JSON or YAML. It doesn't seem like I really need to create these other outputs, but I just want to sanity check that this makes sense... 

Thank you!

Dan

**Comment by verdverm** [comment, marked as answer]
I would use the `cue/load` package like here: https://cuetorials.com/go-api/loading/simple/

If you have the input files as strings in Go already, you can use the `load.Config.Overlay`

The `context.Compile*()` functions takes Options args, one of which is `Scope`. This allows you to provide an existing value as the context or scope for the compilation.

**Reply by gilcrest** [reply]
Wow - that did the trick - thanks a lot! I've been working through your site and it's very helpful. 

One thing I noticed in this example on your site, you have - 

```go
// print the error
fmt.Println("root value:", value)
```

I believe the comment should read:

```go
// print the value
fmt.Println("root value:", value)
```

Anyway, you helped me in 5 minutes what I had spent a good chunk of the night researching - thanks again!

**Reply by gilcrest** [reply]
ok, still struggling a bit here - once I have a `cue.Value`, I can print that value using `fmt.Printf` or `fmt.Println` and it appears to be printing JSON by default? How do I control how `cue.Value` is encoded? If I want it to encode to YAML for instance, is there any easy way to do that? It's not obvious to me from the documentation. It's very easy to do this using the cli and I'm considering just shelling out to the cli instead of trying to figure out the GO API as I'm a bit baffled by it right now. 

If I follow what is happening in the cli, unfortunately, I can only go so far as the encoding that's in use there is within the internal package `cuelang.org/go/internal/encoding` so I don't have access to it...

Any help is really appreciated! 

Basically, I'd like to understand how to easily be able to take a `cue.Value` and be able to generate the same three formats you can with the cli or the playground (JSON, YAML, CUE).

**Reply by verdverm** [reply]
https://cuetorials.com/go-api/basics/printing/
